src/Metadata/MetadataBase.php

```diff
@@ -133,6 +136,22 @@ protected static function getConstraints(): array
         ];
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    protected static function validate(array $data, Collection $constraints): void
+    {
+        static::traitValidate($data, $constraints);
+
+        // The TUF spec requires that all key IDs be unique.
+        // @todo Use Symfony's Unique constraint for this when at least Symfony
+        //   6.1 is required.
+        $keyIds = array_column($data['signatures'], 'keyid');
```

Comment:
@phenaproxima did you consider using `\Symfony\Component\Validator\Constraints\Callback` to a static method so to the we could keep all the config for the metadata validation  in one place. Depending on if we get any usage from other php projects it could be awhile before we can drop support for symfony 5 as it security support till 2025 if this is correct https://endoflife.date/symfony

Reply:
🤔 I didn't even know there was such a constraint.

Reply:
I changed it to use the Callback constraint. Honestly it's not a huge difference because the callback has to be static, but it's a nice clean-up. I could make it a static closure if you'd prefer.